ctrl.c

```diff
@@ -610,6 +610,9 @@ static char *pv_ctrl_get_file_name(const char *path, int buf_index,
 
 	len = path_len - buf_index;
 
+	if (len < 0)
```

Comment:
IMO len and buf_index should be size_t and just check at the beginning if buf_index > path_len

Reply:
Right, I will change it